tests/unit/test_protocols.py

```diff
@@ -181,39 +192,51 @@ def stream(self):
 )
 def test_output_compliance(json_description, case, basename):
     service_description = copy.deepcopy(json_description)
-    operation_name = case.get('name', 'OperationName')
+    operation_name = case.get('given', {}).get('name', 'OperationName')
     service_description['operations'] = {
         operation_name: case,
     }
     case['response']['context'] = {'operation_name': operation_name}
     try:
         model = ServiceModel(service_description)
         operation_model = OperationModel(case['given'], model)
-        parser = PROTOCOL_PARSERS[model.metadata['protocol']](
+        protocol = model.metadata['protocol']
+        parser = PROTOCOL_PARSERS[protocol](
             timestamp_parser=_compliance_timestamp_parser
         )
         # We load the json as utf-8, but the response parser is at the
         # botocore boundary, so it expects to work with bytes.
-        body_bytes = case['response']['body'].encode('utf-8')
-        case['response']['body'] = body_bytes
+        # If a test case doesn't define a response body, set it to `None`.
+        if 'body' in case['response']:
+            body_bytes = case['response']['body'].encode('utf-8')
+            case['response']['body'] = body_bytes
+        else:
+            case['response']['body'] = b''
```

Comment:
Where are we setting the body to `None`? It looks like our fallback is empty bytes. Is this what's causing issues with the xml test above?